File: benchmarks/cross-language/workload.mjs
function checksum() {
  const prime = 1000000007n;
  let numeric = 17n;
  let graph = 29n;
  let text = 7n;
  for (let index = 0n; index < 500000n; index += 1n) {
    numeric = (numeric * 1664525n + index * 1013904223n + 12345n) % prime;
    graph = (graph + ((index * 31n + 17n) % 9973n) * ((index % 13n) + 1n)) % prime;
    text = (text + [84n, 78n, 67n, 72n][Number(index % 4n)]) % prime;
  }
  return (numeric + graph * 31n + text * 131n) % prime;
}

if (checksum() !== 899120682n) {
  console.log("checksum=invalid");
  process.exitCode = 1;
} else {
  console.log("checksum=899120682");
}

File: benchmarks/cross-language/workload.rs
fn checksum() -> u64 {
    const PRIME: u64 = 1_000_000_007;
    let mut numeric = 17_u64;
    let mut graph = 29_u64;
    let mut text = 7_u64;
    for index in 0..500_000_u64 {
        numeric = (numeric * 1_664_525 + index * 1_013_904_223 + 12_345) % PRIME;
        graph = (graph + ((index * 31 + 17) % 9_973) * ((index % 13) + 1)) % PRIME;
        text = (text + match index % 4 {
            0 => 84,
            1 => 78,
            2 => 67,
            _ => 72,
        }) % PRIME;
    }
    (numeric + graph * 31 + text * 131) % PRIME
}

fn main() {
    if checksum() == 899_120_682 {
        println!("checksum=899120682");
    } else {
        println!("checksum=invalid");
        std::process::exit(1);
    }
}
